doc/user-guides/secure-protect-connect.md

```diff
@@ -0,0 +1,372 @@
+
+```
+curl -k -H "'Host: api.${KUADRANT_ZONE_ROOT_DOMAIN}'" "https://$GATEWAY_URL/cars" -i
+
+```
+
+### Define Infrastructure Rate Limiting
+
+So we have a secure communications in place. But there is nothing limiting users from overloading our infrastructure and service components that will sit behind this gateway. Lets and a rate limiting layer to protect our services and infrastructure.
+
+```
+kubectl --context kind-kuadrant-local apply -f - <<EOF
+apiVersion: kuadrant.io/v1beta2
+kind: RateLimitPolicy
+metadata:
+  name: infra-ratelimit
+  namespace: kuadrant-system
+spec:
+  targetRef:
+    name: api-gateway
+    group: gateway.networking.k8s.io
+    kind: Gateway
+  limits:
+    "global":
+      rates:
+      - limit: 5
+        duration: 10
+        unit: second
+EOF
+
+
+kubectl wait ratelimitpolicy infra-ratelimit -n kuadrant-system --for=condition=available
+
+```
+
+The limit here is artificially low in order for us to show it working easily. Lets test it with our endpoint:
+
+```
+for i in {1..10}; do curl -k -H "'Host: api.${KUADRANT_ZONE_ROOT_DOMAIN}'" "https://$GATEWAY_URL/cars" -i; done
+```
+
+Here we should see `409s` start returning after the 5th request.
+
+### Define a Gateway `AuthPolicy`
+
+So communication is secured and we have some protection for our infrastructure, but we do not trust any client to access our endpoints. By default we want to allow only authenticated access. To protect our gateway we will add a `DENY ALL` `AuthPolicy`. Later we will override this with a specific `AuthPolicy` for the API.
+
+```
+kubectl --context kind-kuadrant-local apply -f - <<EOF
+apiVersion: kuadrant.io/v1beta2
+kind: AuthPolicy
+metadata:
+  name: deny-all
+  namespace: kuadrant-system
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: Gateway
+    name: api-gateway
+  rules:
+    authorization:
+      deny-all:
+        opa:
+          rego: "allow = false"
+    response:
+      unauthorized:
+        headers:
+          "content-type":
+            value: application/json
+        body:
+          value: |
+            {
+              "error": "Forbidden",
+              "message": "Access denied by default by the gateway operator. If you are the administrator of the service, create a specific auth policy for the route."
+            }
+EOF
+```
+
+Lets test again. This time we expect a `403`
+
+```
+curl -k -H "'Host: api.${KUADRANT_ZONE_ROOT_DOMAIN}'" "https://$GATEWAY_URL/cars" -i
+
+```
+
+### Define DNSPolicy 
+(skip of you did not setup a DNS provider during the setup)
+
+Now we have our gateway protected and communications secured, we are ready to configure DNS so it is easy for clients to connect and access the APIs we intend to expose via this gateway. Note during the setup of this walk through, we setup a `DNS Provider` secret and a `ManagedZone`.
+
+```
+kubectl --context kind-kuadrant-local apply -f - <<EOF
```

Comment:
I've continued further despite the curl error earlier to see if there's anything else I can validate.
After creating the DNSPolicy, the `wait` times out.
Should there be a ManagedZone created here too?

```
    message: 'reconcile DNSRecords error error reconciling dns records for gateway
      api-gateway: no managed zone for host : *.dm.hcpapps.net'
```

Reply:
Yeah the quick start will set that up but for this specifically you will need to create one yourself